addons/loyalty/data/loyalty_demo.xml

```diff
@@ -0,0 +1,72 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <record id="10_percent_with_code" model="loyalty.program">
+        <field name="name">Code for 10% on orders</field>
+        <field name="program_type">promotion</field>
+        <field name="trigger">with_code</field>
+        <field name="rule_ids" eval="[(0, 0, {
```

Comment:
I think that the rules will be re-created at each update, you should create them separately IMO. 

```
<record id="10_percent_with_code_rule" model="loyalty.rule">
...
```